### PR TITLE
Allow callback add functions to accept args and kwargs

### LINKED ISSUE #6041: Support *args and **kwargs in add_next_tick_callback
This is a minor inconvenience, but it would be nice for `curdoc().add_next_tick_callback` to support `*args` and `**kwargs`

    doc.add_next_tick_callback(func, *args, **kwargs)

Currently the solution is to wrap these in a lambda:

    doc.add_next_tick_callback(lambda: func(*args, **kwargs))

This is low priority, I just thought I'd report it.  This is also what Tornado does:

```python
In [1]: from tornado.ioloop import IOLoop

In [2]: IOLoop.add_callback?
Signature: IOLoop.add_callback(self, callback, *args, **kwargs)
Docstring:
Calls the given callback on the next I/O loop iteration.

It is safe to call this method from any thread at any time,
except from a signal handler.  Note that this is the **only**
method in `IOLoop` that makes this thread-safety guarantee; all
other interaction with the `IOLoop` must be done from that
`IOLoop`'s thread.  `add_callback()` may be used to transfer
control from other threads to the `IOLoop`'s thread.

To add a callback from a signal handler, see
`add_callback_from_signal`.
File:      ~/Software/anaconda/lib/python3.6/site-packages/tornado/ioloop.py
Type:      function
```

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #6041
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

OK I am not super happy with these tests, but I think they are the best that can be added for now absent a refactor of some of the plumbing. (Unless I am missing a different approach)

Ultimately I think some of these private classes can/should be made public so that testing them directly is less unreasonable.

### CODE REVIEW

File: bokeh/server/tests/test_callbacks.py [hunk 1]
```diff
@@ -258,3 +260,36 @@ def cleanup(callback):
...
+        assert list(group._group._periodic_callbacks.keys())[0]() == (10, 20, 30, 40)
+        group.remove_session_callback(cb)
+
+        d = Document()
+        cb = d.add_timeout_callback(func, 1, 10, 20, c=30, d=40)
+
+        group.add_session_callback(cb)
+        assert len(group._group._timeout_callbacks) == 1
+        assert list(group._group._timeout_callbacks.keys())[0]() == (10, 20, 30, 40)
+        group.remove_session_callback(cb)
```
Comment (bryevdv) [inline bokeh/server/tests/test_callbacks.py]: This is the "worst" test. The other tests in `test_document` maintain that a proper `callback_obj` is created. This tries to test that, given a callback obj with args/kwargs, it ultimately stores a function with those parameters baked in.

File: bokeh/tests/test_document.py [hunk 2]
```diff
@@ -543,6 +598,33 @@ def cb(): pass
...
+        assert callback._args == (10, 20)
+        assert callback._kwargs == dict(c=30, d=40)
+
+        callback = d.remove_next_tick_callback(cb)
+        assert len(d.session_callbacks) == 0
+        assert len(events) == 2
+        assert isinstance(events[0], SessionCallbackAdded)
+        assert isinstance(events[1], SessionCallbackRemoved)
+
+
```
Comment (bryevdv) [inline bokeh/tests/test_document.py]: These tests just ensure that a proper "callback obj" is created that reflects any args/kwargs passed in. But it's only about half the journey.

**Comment by bryevdv** [comment]
@mrocklin I added tests, but haven't actually tried things out. Can you give this PR a quick spin?

File: bokeh/document.py [hunk 3]
```diff
@@ -790,6 +799,12 @@ def _add_session_callback(self, callback_obj, callback, one_shot):
                 Whether the callback should immediately auto-remove itself
                 after one execution.
 
+            args (tuple) :
+                Arguments to apply when the callback is invoked
+
+            kwargs (tuple) :
+                Keywoard Arguments to apply when the callback is invoked
```
Comment (canavandl) [inline bokeh/document.py]: spelling "Keyword"

**Pull request closed without merging**